src/core/jit/builder.rs

```diff
@@ -9,26 +9,67 @@ use async_graphql::Positioned;
 use super::model::*;
 use crate::core::blueprint::{Blueprint, Index, QueryField};
 use crate::core::counter::{Count, Counter};
+use crate::core::jit::Variables;
+use crate::core::json::JsonLike;
 use crate::core::merge_right::MergeRight;
 
 pub struct Builder {
     pub index: Index,
     pub arg_id: Counter<usize>,
     pub field_id: Counter<usize>,
     pub document: ExecutableDocument,
+    variables: Variables<async_graphql_value::ConstValue>,
 }
 
 impl Builder {
-    pub fn new(blueprint: &Blueprint, document: ExecutableDocument) -> Self {
+    pub fn new(
+        blueprint: &Blueprint,
+        document: ExecutableDocument,
+        variables: Variables<async_graphql_value::ConstValue>,
```

Comment:
Variables will change with each request but the plan is created only once. Ideally, this information should be used dynamically at the time of execution and not planning - UNLESS the it's always false or always true. We don't need to worry about the "always" cases right now though.